### ISSUE TITLE
Execute script on module condition

### ISSUE THREAD

**Issue by DiegoGuidaF** [Alexays/Waybar#670]
It would be really nice being able to execute custom scripts on a module's condition, for example for the battery widget, if the battery is < 15% send a notification and at 5% suspend the system.

Same can apply for cpu temperature (notify and throttle), ram usage even system time.

Would something like this be easy to implement? This would make a much better system integration without the need of external tools (cron jobs).

**Comment by Alexays** [comment]
Yeah it's something possible with state and should be trivial to implement.
Thanks for the idea!

**Comment by DiegoGuidaF** [comment]
More than happy to help :)
Will be looking forward to the PR so I can review it and help wherever I can!

**Comment by varac** [comment]
Yes please ! Waybar is great, but I had multiple forced shutdowns of my laptop because I didn't notice my low battery :stuck_out_tongue:

**Comment by Iss-in** [comment]
yes please, looking forward to this feature. one more step toward being a complete taskbar

**Comment by ayushvyas-1** [comment]
any update on this?
i really want to have notifications about battery status without running a separate script checking every minute.

**Comment by hboetes** [comment]
Just what I was looking for. I'd also like to get notifications if my battery is running low.

**Comment by DrummyFloyd** [comment]
any update upon this  ?

**Comment by Alexays** [comment]
The custom module supports this on master via `exec-if`: give it a command that acts as a guard (exit 0 to run/show the module, non-zero to hide it) — see the waybar-custom(5) man page for examples. Closing as resolved — reopen if your use case needs something exec-if does not cover.
